### DISCUSSION TITLE
Contributing to this project as a Frontend Developer

### DISCUSSION THREAD

**Discussion by oyerohabib** [shuru-project/shuru#2] [Q&A]
Hi @harshdoesdev. I am opening this discussion concerning your message on creating a documentation site for the project.

**Comment by harshdoesdev** [comment]
Welcome aboard @oyerohabib! For the starters, let's start with discussing the stack for the Docsite. For the framework, we can go with any framework of your choice but my suggestion would be using a framework that supports SSG (e.g. Astro). For hosting the site, you can setup a Github Repository with the initial documentation based on [this](https://dev.to/harshthedev/shuru-a-simple-task-runner-with-built-in-node-version-management-fam) blog I wrote and the README and the examples in this repo. We will later move this project to `docs` folder on the main repository (this repository) and setup the workflows for building and deploying it on Github Pages.

Again, thanks a lot for contributing to this project :rocket:

**Comment by oyerohabib** [comment]
Thank you, @harshdoesdev. I haven't used Astro before actually, however, I am quite flexible with technologies, So I am very much open to learning and using it in no time. I am doing some research about what the best or most preferred technology for this project might be. I will try to document it while making my suggestion. Is there any reason why you perhaps suggest Astro.

**Reply by harshdoesdev** [reply]
Thank you, @oyerohabib! I appreciate your openness to learning new technologies. I mentioned Astro as an example of a Static Site Generator, but the final choice of framework or stack is entirely up to you. I lean toward using a Static Site Generator because it offers benefits like improved SEO and faster loading times. That said, I trust your research and look forward to hearing your recommendation. Please feel free to go with whatever you believe suits the project best!

**Reply by harshdoesdev** [reply]
Hi @oyerohabib, just a heads-up, Shuru now supports Python as well: https://github.com/harshdoesdev/shuru/discussions/3

**Reply by oyerohabib** [reply]
> Thank you, @oyerohabib! I appreciate your openness to learning new technologies. I mentioned Astro as an example of a static site generator, but the final choice of framework or stack is entirely up to you. I lean toward using a static site generator because it offers benefits like improved SEO and faster loading times. That said, I trust your research and look forward to hearing your recommendation. Please feel free to go with whatever you believe suits the project best!

@harshdoesdev Thank you! I am much more familiar with NextJS, which provides SSR, SSG, excellent SEO capabilities, with some excellent features. So, I am trying to see if there's a better need to use another tool. I would like to schedule a call to gain some deeper insights into what the end goal is, and some other clarifications I would like to make. Kindly let me know how we can have this. Thanks for the new update also, that's duly acknowledged.

**Reply by harshdoesdev** [reply]
I’m cool with your choice of NextJS. We can connect via a call on the Shuru Discord server: https://discord.gg/CEcPGRKs
